metadata-ingestion/src/datahub/ingestion/source/superset.py

```diff
@@ -277,15 +358,17 @@ def get_datasource_urn_from_id(self, datasource_id):
 
         if database_id and table_name:
             return make_dataset_urn(
-                platform=platform,
+                platform=platform_instance,
                 name=".".join(
                     name for name in [database_name, schema_name, table_name] if name
                 ),
                 env=self.config.env,
             )
-        return None
+        raise ValueError("Could not construct dataset URN")
```

Comment:
who catches this error? we should degrade gracefully e.g. report an error to the user, continue ingestion

Reply:
I'll reformat this to have a try/except block in the main `emit_dataset_mces` loop, where it logs the raised errors